src/components/NewMessage/NewMessage.vue

```diff
@@ -380,23 +382,46 @@ export default {
 		},
 
 		text(newValue) {
-			this.$store.dispatch('setCurrentMessageInput', { token: this.token, text: newValue })
-
 			// Enable signal sending, only if indicator for this input is on
 			if (this.showTypingStatus) {
+				// Do nothing if join a conversation with pre-filled input
+				if (this.text === this.$store.getters.currentMessageInput(this.token)) {
+					return
+				}
+
+				// Reset indicator for empty input
 				if (!newValue) {
 					this.resetTypingIndicator()
 					return
 				}
 
-				if (!this.typingTimeout) {
-					this.typingTimeout = setTimeout(() => {
-						this.resetTypingIndicator()
-					}, 5000)
-					this.$store.dispatch('setTyping', { typing: true })
-				}
+				this.typingLastActivity = new Date().getTime()
 
+				// Send typing status, if input was changed between 1s-10s at each 10s interval
+				if (!this.typingInterval) {
+					this.typingPrevActivity = new Date().getTime()
+
+					// Send first signal after 1 second of typing
+					setTimeout(() => {
+						if (this.typingLastActivity - this.typingPrevActivity > 0) {
+							this.$store.dispatch('setTyping', { typing: true })
+							this.typingPrevActivity = new Date().getTime()
+						}
+					}, 1000)
+
+					// Continuously send signals with 10s interval if still typing
+					this.typingInterval = setInterval(() => {
+						if (this.typingLastActivity - this.typingPrevActivity > 0) {
+							this.$store.dispatch('setTyping', { typing: true })
+							this.typingPrevActivity = new Date().getTime()
+						} else {
+							this.resetTypingIndicator()
+						}
+					}, 10000)
```

Comment:
In which case do we have `this.typingLastActivity - this.typingPrevActivity <= 0`?